File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        a=[]
        def preorder(node,a):
            if node:
                a.append(node.val)
                preorder(node.left,a)
                preorder(node.right,a)
        preorder(root,a)
        return a

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List <Integer> a=new ArrayList<Integer>();
        preorder(root,a);
        return a;
    }
    static void preorder(TreeNode node,List<Integer> a){
        if(node==null){
            return;
        }
        a.add(node.val);
        preorder(node.left,a);
        preorder(node.right,a);
    }
}